File: chapter-11-greedy/곱하기-혹은-더하기-book.cpp
#include <iostream>

using namespace std;

int solution(string S){
    // 첫 번째 문자를 숫자로 변경한 값을 대입
    long long result = S[0] - '0';

    for (int i = 1; i < S.size(); ++i) {
        // 두 수 중에서 하나라도 '0' 혹은 '1'인 경우, 곱하기보다는 더하기 수행
        int num = S[i] - '0';
        if (num <= 1 || result <= 1) {
            result += num;
        }
        else {
            result *= num;
        }
    }
    return result;
} 

int main(){
    string S;
    cin >> S;

    int result = solution(S);
    cout << "실행 결과: " << result << endl;
    return 0;
}

File: chapter-11-greedy/곱하기-혹은-더하기-book.py
import time

def solution(S):
    # 첫 번째 문자를 숫자로 변경하여 대입
    result = int(S[0])

    for i in range(1, len(S)):
        # 두 수중에서 하나라도 '0' 혹은 '1'인 경우, 곱하기보다는 더하긱 수행
        num = int(S[i])
        if num <= 1 or result <= 1:
            result += num
        else :
            result *= num

    return result 


S = input()
start_time = time.time()
result = solution(S)
end_time = time.time()

print('실행 결과:', result)
print('걸린 시간:', end_time - start_time)
